src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -587,15 +587,15 @@ private static byte[] getJ0(byte[] iv, byte[] subkeyH, int blockSize) {
      */
     private static int implGCMCrypt(byte[] in, int inOfs, int inLen, byte[] ct,
                                     int ctOfs, byte[] out, int outOfs,
-                                    GCTR gctr, GHASH ghash) {
+                                    GCTR gctr, GHASH ghash, boolean encryption) {
```

Comment:
It looks to me that you don't need to introduce this "boolean encryption" here as it is simply (ct == out), which can easily be calculated in the intrinsics and that saves a lot of code change.

Reply:
@ferakocz Thank you for your comment. I will make the change.

Reply:
@ascarpino Apologies for the delay in responding, I was away on vacation. There are fewer number of registers available in the AVX2 algorithm as compared to AVX512. That's why its essential for the intrinsic to know if it is encryption or decryption this time around. I will be implementing Ferenc's suggestion and remove the boolean variable.